plugin/header/header_test.go

```diff
@@ -80,3 +80,22 @@ func TestHeader(t *testing.T) {
 		}
 	}
 }
+
+func TestHeaderQueryRules(t *testing.T) {
+	wr := dnstest.NewRecorder(&test.ResponseWriter{})
+	next := plugin.HandlerFunc(func(ctx context.Context, writer dns.ResponseWriter, msg *dns.Msg) (int, error) {
+		if !msg.RecursionDesired {
```

Comment:
Why not add a table driven test case like we are doing for `TestHeaderResponseRules` for consistency.

Reply:
Sure. I initially tried it to keep the consistency. Sadly I have no real idea how the test could be written in this table style. I tried to implement some kind of nextAssert function which performs the assertion but I didn't manage it to make the compiler happy. As mentioned this are literally my first Go steps. I would highly appreciate some help here.

Reply:
This [article](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) is great for introduction. We have more [test cases](https://github.com/coredns/coredns/blob/master/plugin/rewrite/name_test.go#L35) this like in `rewrite` plugin. Please let me know if it doesn't help?

Reply:
Thank you very much for your help. Especially the test case that you wrote helped me a lot. I honestly didn't thought about checking the DNSMsg after passing it through the handler function. 🤦‍♂️  My brain stuck a bit too much on Go function arguments are Pass-by-Value and forgot about using Pass-by-Reference in this case.

Reply:
Glad I was able to help! :+1: